db/patch12.sql

```diff
@@ -0,0 +1,10 @@
+-- You should not modify if this have pushed to Github, unless it does serious wrong with the db.
+BEGIN TRANSACTION;
+
+ALTER TABLE user
+    ADD twofa_secret VARCHAR(64);
+
+ALTER TABLE user
```

Comment:
`NOT NULL` would be better

Reply:
I actually think we shouldn't add `NOT NULL`, since there is no default value we could give it. It'll be only filled if it's necessary like when 2FA gets prepared.

Reply:
👍 Got it, haven't deeply test it. Just be careful that if it is a `null`,  Both `(twofa_status == 0)` and `(twofa_status == 1)`  are `false`

Reply:
Are you referring to the column `twofa_secret` or `twofa_status`?
`twofa_status` gets the default value 0 by default and if 2FA is enabled it's 1 and is the main indicator if 2FA is enabled.
`twofa_secret` is only there to store the secret.

Reply:
Yes, `twofa_status`.

Reply:
So bad Github was highlighted Line 4 in the comment thread. I am actually commenting on Line 7, lol.

Reply:
Ohh Alright! xD
Yea agree on that and just added it with [`14b7688`](https://github.com/louislam/uptime-kuma/pull/363/commits/14b7688b70445392aa4a341705d13fc1e1a4a4f5)